scanner/fetcher.go

```diff
@@ -108,16 +113,23 @@ func (f *Fetcher) Prepare(ctx context.Context) (*ct.SignedTreeHead, error) {
 	return sth, nil
 }
 
-// Run performs fetching of the Log. Blocks until scanning is complete or
-// context is cancelled. For each successfully fetched batch, runs the fn
-// callback.
+// Run performs fetching of the Log. Blocks until scanning is complete, the
+// passed in context is canceled, or Stop is called. For each successfully
+// fetched batch, runs the fn callback.
 func (f *Fetcher) Run(ctx context.Context, fn func(EntryBatch)) error {
 	glog.V(1).Info("Starting up Fetcher...")
 	if _, err := f.Prepare(ctx); err != nil {
 		return err
 	}
 
-	ranges := f.genRanges(ctx)
```

Comment:
Need a comment here, otherwise it's too hard to spot the difference in behaviour between `ctx` and `cctx` -- something like:
```golang
// Use a separately-cancellable context for the range generator, so we
// we can close it down (in Stop) but still let the fetchers below run to
// completion.
```

Reply:
Done.